com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/StepRequestHandler.java

```diff
@@ -438,15 +439,19 @@ private boolean shouldDoExtraStepInto(int originalStackDepth, Location originalL
         return true;
     }
 
-    private boolean isSameLocation(Location original, Location current) {
+    private boolean isSameLocation(Location original, Location current, MethodInvocation targetStepIn) {
         if (original == null || current == null) {
             return false;
         }
 
         Method originalMethod = original.method();
         Method currentMethod = current.method();
+        // if the lines doesn't match, check if the current line is still behind the
+        // target if a target exist. This handles where the target is part of a
+        // expression which is wrapped.
         return originalMethod.equals(currentMethod)
-            && original.lineNumber() == current.lineNumber();
+                && (original.lineNumber() == current.lineNumber()
+                        || (targetStepIn != null && targetStepIn.lineStart > current.lineNumber()));
```

Comment:
The last condition check should be `(targetStepIn != null && targetStepIn.lineEnd >= current.lineNumber())`, which supports multiline expression better.